### ISSUE TITLE
Integrate into the core

### ISSUE THREAD

**Issue by nicoddemus** [pytest-dev/pytest-subtests#71]
Opening this to debate if it is time to integrate this into the core.

Pros to integrate it into the core:

* Adds support for a standard `unittest` feature.
* A different (arguably simpler) way to execute parametrized tests via the `subtests` fixture.
* Some features (https://github.com/pytest-dev/pytest-subtests/issues/6 for example) are hard to implement as plugin without further changes into the core.

Cons:

* Yet another maintenance burden to the core.

Opinions?

cc @bluetech @RonnyPfannschmidt @asottile @The-Compiler

**Comment by RonnyPfannschmidt** [comment]
If lastfailed/stepwise support is fixed im all for integration

**Comment by nicoddemus** [comment]
Yes, I would think that in order to integrate into the core, we would also take the opportunity of fixing some standing issues like lastfailed/stepwise.

**Comment by RonnyPfannschmidt** [comment]
@nicoddemus should we release a code removal + deprecated trove classifier of this that only requires the modern pytest that includes it + a import alias deprecation

**Comment by nicoddemus** [comment]
Not sure, this will break those that depend on the plugin today. Once pytest 9 is released and installed alongside this plugin, pytest itself will show a warning via https://github.com/pytest-dev/pytest/blob/e3facc0f8115a52606151c436ab3cb0816dbc0c3/src/_pytest/deprecated.py#L28.
